compiler/rustc_session/src/session.rs

```diff
@@ -467,6 +467,7 @@ impl Session {
         feature: Symbol,
     ) -> DiagnosticBuilder<'a, ErrorGuaranteed> {
         let mut err = self.parse_sess.create_err(err);
+        err.code = std::option::Option::Some(error_code!(E0658));
```

Comment:
I think it'd be a good idea to have a check here to make sure `err.code` isn't already set - adding a code to an error that doesn't have on at all is one thing, changing the code of an error might be more confusing.

Reply:
Added in latest hotfix commit